server/datastore/psql/maintenance_requests.go

```diff
@@ -0,0 +1,51 @@
+package psql
+
+import (
+	"time"
+
+	"github.com/a2fumn2022/gpmn-nsl/server/datastore"
+)
+
+// Import the necessary packages
+type rawTime []byte
+
+func (t rawTime) Time() (time.Time, error) {
+	return time.Parse("15:04:05", string(t))
+}
+
+func (db *DB) AllMaintenanceRequests() []datastore.MaintenanceRequest {
+
+	rows, err := db.connection.Query("select * from public.nsl_maintenance LIMIT $1", 777) //Number limits # of requests, can be anything
+	if err != nil {
+		// handle this error better than this
+		panic(err)
```

Comment:
Rather than throwing a panic, you should return an error. See `CreateMaintenanceRequest` to see how that is handled.

Reply:
^ yeah you can change the function signature to return 2 things rather than just the list of MaintenanceRequests.

ie: `func (db *DB) AllMaintenanceRequests() ([]datastore.MaintenanceRequest, error) { ... }`